src/proto.rs

```diff
@@ -1149,23 +1157,30 @@ impl Decoder for WebsocketProtocol {
             }
 
             if payload_length == 126 {
-                ensure_buffer_has_space!(src, 4);
+                ensure_buffer_has_space!(src, offset + 2);
                 // SAFETY: The ensure_buffer_has_space call has validated this
                 // A conversion from two u8s to a u16 cannot fail
                 payload_length = u16::from_be_bytes(unsafe {
                     src.get_unchecked(2..4).try_into().unwrap_unchecked()
                 }) as usize;
+                if payload_length <= 125 {
+                    return Err(Error::Protocol(ProtocolError::InvalidPayloadLength));
+                }
                 offset = 4;
             } else if payload_length == 127 {
-                ensure_buffer_has_space!(src, 10);
+                ensure_buffer_has_space!(src, offset + 8);
                 // SAFETY: The ensure_buffer_has_space call has validated this
                 // A conversion from 8 u8s to a u64 cannot fail
                 payload_length = u64::from_be_bytes(unsafe {
                     src.get_unchecked(2..10).try_into().unwrap_unchecked()
                 }) as usize;
+                if payload_length <= u16::MAX as usize {
```

Comment:
clippy suggests `u16::try_from(payload_length).is_ok()` here, but that's inconsistent with the `Encoder`'s `chunk_size > u16::MAX as usize`. I think both or neither should change

Reply:
I think we should just listen to clippy here and change both occurences